src/uct/ib/base/ib_device.c

```diff
@@ -350,6 +350,15 @@ const uct_ib_device_spec_t* uct_ib_device_spec(uct_ib_device_t *dev)
                     default settings for unknown devices */
 }
 
+static size_t uct_ib_device_get_ib_gid_index(uct_ib_md_t *md)
```

Comment:
maybe better set correct `md->config.gid_index` in `uct_ib_md_open`?

Reply:
i cant set it there since the gid_index should be set per ib iface, not md. since mds can be shared.